lua/nvim-tree.lua

```diff
@@ -125,10 +153,15 @@ function M.find_file(with_open, bufnr)
     M.open()
   end
 
+  -- if we don't schedule, it will search for NvimTree
   vim.schedule(function()
-    -- if we don't schedule, it will search for NvimTree
-    if _config.update_focused_file.update_cwd then
+    -- bang forces it to run M.change_root
+    if bang then
+      M.change_root(filepath)
+    elseif _config.update_focused_file.update_cwd then
```

Comment:
`update_focused_file.update_cwd` and `update_focused_file.update_root` are actually the same except for `root_dirs`.

We can remove `update_focused_file.update_root` and just have `update_focused_file.update_cwd` use `root_dirs`.

`update_base_dir_with_filepath` can be replaced with `change_root` as they are doing the same thing.

 `change_root` needs to be modified to honor `update_focused_file.ignore_list`.

Are you OK with that @kyazdani42 ?

Reply:
I think `update_focused_file.update_root` might be a better name here. Why not keep it and mark `update_focused_file.update_cwd` as deprecated?

Reply:
Thank you.